src/redis_index.c

```diff
@@ -484,7 +484,8 @@ int Redis_DropIndex(RedisSearchCtx *ctx, int deleteDocuments) {
 
   SchemaPrefixes_RemoveSpec(spec);
 
-  if (deleteDocuments || !!(spec->flags & Index_Temporary)) {
+  // keys are deleted from slaves automatically
+  if (isMaster && (deleteDocuments || !!(spec->flags & Index_Temporary))) {
```

Comment:
I do not think this is a right fix, master should just not replicate DD or replicate KEEPDOCS always. It should be the case on temporary indexes as well, any idea why we had an exception on temporary indexes?

Reply:
AFAICR, the behavior we decided on is that temporary indexes delete all documents attached to them.
During `FT.CREATE` on a temporary index, we set on both master and slave `RM_CreateTimer` with a callback that calls `Redis_DropIndex`.
Therefore both shards might call RM_Call("DEL") independently but we want the behavior to be different. For master, we want to call "DEL" locally and propagate it to all slaves, and on a slave, to do nothing.

I don't see how we can bridge this otherwise.